src/main/java/build/buildfarm/server/BuildFarmServer.java

```diff
@@ -136,7 +136,11 @@ private static MetricsPublisher getMetricsPublisher(MetricsConfig metricsConfig)
     }
   }
 
-  public void start(String publicName) throws IOException {
+  public synchronized void start(String publicName) throws IOException {
+    if (stopping) {
```

Comment:
This should just be an invalid state - start should not be callable in any context where stopping was able to be set: prefer guava's `checkState` here.

Reply:
Good point, will fix!